src/expression/expression-builder.ts

```diff
@@ -54,6 +55,68 @@ import { CaseNode } from '../operation-node/case-node.js'
 import { isUndefined } from '../util/object-utils.js'
 
 export interface ExpressionBuilder<DB, TB extends keyof DB> {
+  /**
+   * Creates a binary expression.
+   *
+   * This function returns an {@link Expression} and can be used pretty much anywhere.
+   * See the examples for a couple of possible use cases.
+   *
+   * ### Examples
+   *
+   * In the following example `eb` is used to increment an integer column:
+   *
+   * ```ts
+   * db.updateTable('person')
+   *   .set((eb) => ({
+   *     age: eb('age', '+', 1)
+   *   }))
+   *   .where('id', '=', id)
+   * ```
+   *
+   * The generated SQL (PostgreSQL):
+   *
+   * ```sql
+   * update "person"
+   * set "age" = "age" + $1
+   * where "id" = $2
+   * ```
+   *
+   * By default the third argument is a value. {@link ref} can be used to
+   * pass in a column reference instead:
+   *
+   * ```ts
+   * db.updateTable('person')
+   *   .set((eb) => ({
+   *     age: eb('age', '+', eb.ref('age'))
+   *   }))
+   *   .where('id', '=', id)
+   * ```
+   *
+   * The generated SQL (PostgreSQL):
+   *
+   * ```sql
+   * update "person"
+   * set "age" = "age" + "age"
+   * where "id" = $1
+   * ```
+   */
+  <RE extends ReferenceExpression<DB, TB>, OP extends BinaryOperatorExpression>(
+    lhs: RE,
+    op: OP,
+    rhs: OperandValueExpressionOrList<DB, TB, RE>
+  ): ExpressionWrapper<
+    DB,
+    TB,
+    OP extends ComparisonOperator
+      ? SqlBool
+      : ExtractTypeFromReferenceExpression<DB, TB, RE>
+  >
+
+  /**
+   * Returns a copy of `this` expression builder.
+   */
```

Comment:
```suggestion
  /**
   * Returns a copy of `this` expression builder, for destructuring purposes.
   * 
   * ### Examples
   *
   * ```ts
   * db.selectFrom('person')
   *   .where(({ eb, exists, selectFrom }) =>
   *     eb('first_name', '=', 'Jennifer').and(
   *       exists(selectFrom('pet').whereRef('owner_id', '=', 'person.id').select('pet.id'))
   *     )
   *   )
   *   .selectAll()
   * ```
   * 
   * The generated SQL (PostgreSQL):
   *
   * ```sql
   * select * from "person" where "first_name" = $1 and exists (
   *   select "pet.id" from "pet" where "owner_id" = "person.id"
   * )
   * ```
   */
```

Reply:
> A small concern might be the addition of DB and TB to ExpressionWrapper's interface, and how it will affect consumers who make their own helpers. Is it still as flexible as before? I guess we'll find out and adjust if there are issues.

I don't think `ExpressionWrapper` is especially useful in customer code. `Expression` and `AliasableExpression` are still simple.